tests/lib/__init__.py

```diff
@@ -542,19 +534,38 @@ def run(self, *args, **kw):
                 )
             allow_stderr_warning = True
 
+        if allow_stderr_error:
+            if allow_stderr_warning is not None and not allow_stderr_warning:
+                raise RuntimeError(
+                    'cannot pass allow_stderr_warning=False with '
+                    'allow_stderr_error=True'
+                )
+
+        # Default values if not set.
+        if allow_stderr_error is None:
+            allow_stderr_error = False
+        if allow_stderr_warning is None:
+            allow_stderr_warning = allow_stderr_error
+
         # Pass expect_stderr=True to allow any stderr.  We do this because
         # we do our checking of stderr further on in check_stderr().
         kw['expect_stderr'] = True
         result = super(PipTestEnvironment, self).run(cwd=cwd, *args, **kw)
 
-        check_stderr(
+        if expect_error:
```

Comment:
Actually, I just noticed it would be good if this option were documented in the docstring along with the others, including explaining its relationship to the others (what it implies), especially because you’re changing how it behaves.

Reply:
How does this sound?

```
        :param expect_error: if False (the default), asserts that the command
            exits with 0.  Otherwise, asserts that the command exits with a
            non-zero exit code.  Passing True also implies allow_stderr_error
            and allow_stderr_warning.
```

Reply:
Great!

Reply:
Updated [here](https://github.com/pypa/pip/pull/6856/files#diff-a54fadd4f116c7810b032d445a9842b3R496-R499).